godaddy-wordpress/sake
# Add `bundle` task to bundle plugin frontend from a npm package

## Summary

This PR will add a new `bundle` task which will look into a configuration property from the local plugin `sake.config.js` configuration file, which might contain `bundle.scripts` and `bundle.styles`:

```javascript
module.exports = {
	framework: 'v5',
	bundle: {
        scripts:
		[{
			source: '@gdcorp-partners/woocommerce-gateway-authorize-net-cim/dist/js',
			file: 'app.js',
			destination: 'assets/js/blocks',
		}],
		styles:
		[{
			source: '@gdcorp-partners/woocommerce-gateway-authorize-net-cim/dist/js',
			file: 'app.css',
			destination: 'assets/css/blocks',
		}]
	}
}
```

- The `source` should match the corresponding package path expected to be installed by `npm install` (thus, the corresponding plugin should also have it in its package.json, pointing to a specific version used by the plugin). Note that the package name specified in `package.json` and the actual path could differ: `source` should point to the path.
- The `file` property is the file name inside the corresponding `source` folder in `node_modules`.
- The `destination` should tell Sake where to copy the `file` in the local plugin path (there is no default).

## Details

Other than `bundle.scripts` and `styles` we can have in the future additional items to bundle, for example `bundle.i18n` if we decide to bundle language packs that are generated in another package, etc.  [update: looks like we won't need i18n but instead update the pomo tools to wp cli]

## QA

Changes pertaining to some potfile generation were already reviewed in https://github.com/godaddy-wordpress/sake/pull/73

To test this branch, [follow this guide](https://github.com/godaddy-wordpress/sake/wiki/Using-a-development-version-of-Sak%C3%A9) to use a dev copy of Sake and test it out with Authorize.Net or other plugin that has an external script dependency. You need to make sure that that dependency is also added to `package.json` so that the `sake bundle` task will install it with `npm install`. 

[Follow these instructions](https://godaddy-corp.atlassian.net/wiki/spaces/MWCENG/pages/53150095/Javascript+Setup) to make sure npm can install packages from private sources.

See https://github.com/gdcorp-partners/woocommerce-gateway-authorize-net-cim/pull/91 for a demo PR
